src/java.desktop/share/classes/sun/print/PrintJobDelegate.java

```diff
@@ -0,0 +1,1233 @@
+             * left of the imageable area when using the new "printable"
+             * origin attribute, otherwise its the physical origin (for
+             * backwards compatibility. We emulate this by creating
+             * a PageFormat which matches and then performing the
+             * translate to the origin. This is a no-op if physical
+             * origin is specified.
+             */
+            currentGraphics.translate(pageFormat.getImageableX(),
+                                      pageFormat.getImageableY());
+
+            /* Scale to accommodate AWT's notion of printer resolution */
+            double awtScale = 72.0/getPageResolutionInternal();
+            currentGraphics.scale(awtScale, awtScale);
+
+            /* The caller wants a Graphics instance but we do
+             * not want them to make 2D calls. We can't hand
+             * back a Graphics2D. The returned Graphics also
+             * needs to implement PrintGraphics, so we wrap
+             * the Graphics2D instance. The PrintJob API has
+             * the application dispose of the Graphics so
+             * we create a copy of the one returned by PrinterJob.
+             */
+            //printGraphics = new ProxyPrintGraphics(currentGraphics.create(), this);
+            printGraphics =  currentGraphics.create();
+        }
+
+        return printGraphics;
+    }
+
+    /**
+     * Returns the dimensions of the page in pixels.
+     * The resolution of the page is chosen so that it
+     * is similar to the screen resolution.
+     * Except (since 1.3) when the application specifies a resolution.
+     * In that case it is scaled accordingly.
+     */
+    public Dimension getPageDimension() {
+        double wid, hgt, scale;
+        if (pageAttributes != null &&
+            pageAttributes.getOrigin()==OriginType.PRINTABLE) {
+            wid = pageFormat.getImageableWidth();
+            hgt = pageFormat.getImageableHeight();
+        } else {
+            wid = pageFormat.getWidth();
+            hgt = pageFormat.getHeight();
+        }
+        scale = getPageResolutionInternal() / 72.0;
+        return new Dimension((int)(wid * scale), (int)(hgt * scale));
+    }
+
+     private double getPageResolutionInternal() {
+        if (pageAttributes != null) {
+            int []res = pageAttributes.getPrinterResolution();
+            if (res[2] == 3) {
+                return res[0];
+            } else /* if (res[2] == 4) */ {
+                return (res[0] * 2.54);
+            }
+        } else {
+            return 72.0;
+        }
+    }
+
+    /**
+     * Returns the resolution of the page in pixels per inch.
+     * Note that this doesn't have to correspond to the physical
+     * resolution of the printer.
+     */
+    public int getPageResolution() {
+        return (int)getPageResolutionInternal();
+    }
+
+    /**
+     * Returns true if the last page will be printed first.
+     */
+    public boolean lastPageFirst() {
```

Comment:
It appears to be unused.

Reply:
because the method just returns false the delegate wasn't needed.
I decided to make it call the delegate anyway for consistency so now it is used.